core/src/test/java/com/google/errorprone/refaster/UMethodInvocationTest.java

```diff
@@ -41,7 +41,8 @@ public void match() {
     ULiteral oneLit = ULiteral.intLit(1);
     ULiteral barLit = ULiteral.stringLit("bar");
     UMethodInvocation invocation =
-        UMethodInvocation.create(fooIdent, ImmutableList.<UExpression>of(oneLit, barLit));
+        UMethodInvocation.create(
```

Comment:
Should we extend these tests or leave it as is?

Reply:
The unification logic does not consider the type parameters (should it?), so I guess right now there's nothing to test. Maybe we can use the varargs overload instead:
```java
UMethodInvocation.create(fooIdent, oneLit, barLit);
```
(But not sure that makes the test easier to understand :thinking:.)

That said, perhaps we could add an extra inline test below, e.g. one that handles two type arguments (since the Refaster test further down already covers the one-type argument case).

Reply:
I tried adding the `typeArguments` to the unification in `UMethodInvocation`. However, the [ImplicitTypesInlinedTemplate](https://github.com/google/error-prone/blob/master/core/src/test/java/com/google/errorprone/refaster/testdata/template/ImplicitTypesInlinedTemplate.java) fails after doing so, specifically on [this example](https://github.com/google/error-prone/blob/master/core/src/test/java/com/google/errorprone/refaster/testdata/input/ImplicitTypesInlinedTemplateExample.java#L32).
The unification fails simply because the expression in the `@BeforeTemplate` does not have type arguments, but the matched code does.

We could try to define the semantics in such a way that the unification process also accounts for implicit type arguments, but it feels like a bridge too far for this PR. So I would suggest to leave it as is. Let me know what you think.